Userland/Libraries/LibAudio/ConnectionToServer.cpp

```diff
@@ -27,8 +27,10 @@ ConnectionToServer::ConnectionToServer(NonnullOwnPtr<Core::Stream::LocalSocket>
         Core::EventLoop enqueuer_loop;
         m_enqueuer_loop = &enqueuer_loop;
         enqueuer_loop.exec();
-        Threading::MutexLocker const locker(m_enqueuer_loop_destruction);
```

Comment:
If we want to be super defensive, this `construct` is actually unsafe.

```
    NonnullRefPtr<Threading::Thread> m_background_audio_enqueuer;
    Core::EventLoop* m_enqueuer_loop;
    Threading::Mutex m_enqueuer_loop_destruction;
```

The  `m_enqueuer_loop` pointer is initialized *after* `m_background_audio_enqueuer`.

If the OS decides (it won't decide but let's have a thought experiment) to schedule the new thread immediately upon construction and not schedule the original thread until we get to the loop.exec() call, we're in trouble.

So this sequence is legal:

```
Thread::construct()
                                      EventLoop::EventLoop()
                                      m_enqueuer_event_loop = &event_loop
m_enqueuer_loop = nullptr
```

The fix is simple though. assign to m_background_audio_enqueuer in the constructor body, after the init list runs. if you had tried to use the mutex in the thread as well before doing some long running blocking operation, it would be possible to try to use a mutex that hasn't been initialized, as well. This kind of foot-gun is why I kind of like explicitly initializing members in the init list, but alas, that's not our style :)

Reply:
The background audio enqueuer thread is not started in the initializer list or the constructor. This means it cannot run while the constructor is running, at the earliest it will run upon an immediate call to `async_enqueue`. Luckily our Thread APIs don't immediately start threads :^)